src/LinAlg/hiopMatrixSparseCsrCuda.cpp

```diff
@@ -301,11 +298,8 @@ void hiopMatrixSparseCSRCUDA::addDiagonal(const double& alpha, const hiopVector&
 {
   assert(nrows_ == D.get_size());
   assert(nrows_ == ncols_ && "Matrix must be square");
-#ifdef HIOP_USE_RAJA
-  assert(dynamic_cast<const hiopVectorCuda*>(&D) && "input vector must be RAJA-CUDA");
-#else
-  assert("input vector must be RAJA-CUDA, but HiOp was not built with RAJA");
-#endif
+  assert(dynamic_cast<const hiopVectorCuda*>(&D) && "input vector must be CUDA");
```

Comment:
I tried to figure this out from other LinAlg objects, but couldn't get that far - should we place any of these asserts behind `#ifdef DEEPCHECKS`, or are they necessary to have during all HiOp execution?

Reply:
they catch a supposedly "no-go" case caused by a potential change in the code by a developer (use of improper HiOp vector type), so it should pop up under debug builds. It is not user-related nor potentially some abnormal numerics, which sometimes go under deepchecks.